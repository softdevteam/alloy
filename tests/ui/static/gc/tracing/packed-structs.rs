#![deny(misaligned_gc_pointers)]
#![feature(gc)]

struct RawPtr(*mut u8);

#[repr(packed)]
struct PackedUsize { //~ ERROR: contains a non-word aligned field `usize`
    x: u16,
    y: usize,
}

#[repr(packed)]
struct PackedUsizeAligned {
    x: u16,
    y: u16,
    z: u32,
    a: usize,
}

#[repr(packed)]
struct PackedI64 { //~ ERROR: contains a non-word aligned field `i64`
    x: u16,
    y: i64,
}

#[repr(packed)]
struct PackedU64 { //~ ERROR: contains a non-word aligned field `u64`
    x: u16,
    y: u64,
}

#[repr(packed)]
struct PackedRef<'a> { //~ ERROR: contains a non-word aligned field `&'a usize`
    x: u16,
    y: &'a usize,
}

#[repr(packed)]
struct PackedRefAligned<'a> {
    x: u16,
    y: u16,
    z: u32,
    a: &'a usize,
}

#[repr(packed)]
struct PackedRaw { //~ ERROR: contains a non-word aligned field `*mut usize`
    x: u16,
    y: *mut usize,
}

#[repr(packed)]
struct PackedRawAligned {
    x: u16,
    y: u16,
    z: u32,
    a: *mut usize,
}

#[repr(packed)]
struct PackedTuple { //~ ERROR: contains a non-word aligned field `usize`
    x: u16,
    y: (usize, u8),
}

#[repr(packed)]
struct PackedArray { //~ ERROR: contains a non-word aligned field `usize`
    x: u16,
    y: [usize; 3],
}

#[repr(packed)]
struct PackedArrayOffset { //~ ERROR: contains a non-word aligned field `u64`
    x: usize,
    y: [u16; 3],
    z: u64
}

#[repr(packed)]
struct PackedAdt { //~ ERROR: contains a non-word aligned field `*mut u8`
    x: u16,
    y: RawPtr,
}

fn main() {}
