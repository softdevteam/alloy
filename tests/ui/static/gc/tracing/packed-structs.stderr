error: contains a non-word aligned field `usize`
  --> $DIR/packed-structs.rs:7:1
   |
LL | / struct PackedUsize {
LL | |     x: u16,
LL | |     y: usize,
LL | | }
   | |_^
   |
note: the lint level is defined here
  --> $DIR/packed-structs.rs:1:9
   |
LL | #![deny(misaligned_gc_pointers)]
   |         ^^^^^^^^^^^^^^^^^^^^^^

error: contains a non-word aligned field `i64`
  --> $DIR/packed-structs.rs:21:1
   |
LL | / struct PackedI64 {
LL | |     x: u16,
LL | |     y: i64,
LL | | }
   | |_^

error: contains a non-word aligned field `u64`
  --> $DIR/packed-structs.rs:27:1
   |
LL | / struct PackedU64 {
LL | |     x: u16,
LL | |     y: u64,
LL | | }
   | |_^

error: contains a non-word aligned field `&'a usize`
  --> $DIR/packed-structs.rs:33:1
   |
LL | / struct PackedRef<'a> {
LL | |     x: u16,
LL | |     y: &'a usize,
LL | | }
   | |_^

error: contains a non-word aligned field `*mut usize`
  --> $DIR/packed-structs.rs:47:1
   |
LL | / struct PackedRaw {
LL | |     x: u16,
LL | |     y: *mut usize,
LL | | }
   | |_^

error: contains a non-word aligned field `usize`
  --> $DIR/packed-structs.rs:61:1
   |
LL | / struct PackedTuple {
LL | |     x: u16,
LL | |     y: (usize, u8),
LL | | }
   | |_^

error: contains a non-word aligned field `usize`
  --> $DIR/packed-structs.rs:67:1
   |
LL | / struct PackedArray {
LL | |     x: u16,
LL | |     y: [usize; 3],
LL | | }
   | |_^

error: contains a non-word aligned field `u64`
  --> $DIR/packed-structs.rs:73:1
   |
LL | / struct PackedArrayOffset {
LL | |     x: usize,
LL | |     y: [u16; 3],
LL | |     z: u64
LL | | }
   | |_^

error: contains a non-word aligned field `*mut u8`
  --> $DIR/packed-structs.rs:80:1
   |
LL | / struct PackedAdt {
LL | |     x: u16,
LL | |     y: RawPtr,
LL | | }
   | |_^

error: aborting due to 9 previous errors

